hack/markdown-preprocess

```diff
@@ -53,6 +56,36 @@ class Preprocessor():
         os.chmod(outfile_tmp, 0o444)
         os.rename(outfile_tmp, outfile)
 
+    def track_optionfile(self, optionfile: str):
+        """
+        Keep track of which man pages use which option files
+        """
+        if optionfile not in self.used_by:
+            self.used_by[optionfile] = []
+        self.used_by[optionfile].append(self.podman_subcommand('full'))
+
+    def rewrite_optionfiles(self):
+        """
+        Rewrite all option files, such that they include header comments
+        cross-referencing all the man pages in which they're used.
+        """
+        for optionfile in self.used_by:
+            tmpfile = optionfile + '.tmp'
+            with open(optionfile, 'r') as fh_in, open(tmpfile, 'w') as fh_out:
+                fh_out.write("####> This option file is used in:\n")
+                used_by = ', '.join(x for x in self.used_by[optionfile])
+                fh_out.write(f"####>   podman {used_by}\n")
+                fh_out.write("####> If you edit this file, make sure your changes\n")
+                fh_out.write("####> are applicable to all of those.\n")
+                for line in fh_in:
+                    if not line.startswith('####>'):
+                        fh_out.write(line)
+            # Compare files; only rewrite if the new one differs
+            if not filecmp.cmp(optionfile, tmpfile):
+                os.rename(tmpfile, optionfile)
```

Comment:
Why this extra check? Can't we just rename always?

Reply:
It's just a habit of mine: `ls -lt | head -20` is easily in my top ten most frequent commands list (aliased, of course). It's incredibly useful for investigating all sorts of failures and problems and changes. So I'm always careful not to pollute directories with unnecessary mtime changes.